src/app/level2/level2.component.ts

```diff
@@ -5,13 +5,89 @@ import { Component } from '@angular/core';
   styleUrls: ['./level2.component.css'],
 })
 export class Level2Component {
-  public currentPlayerIndex = 1;
+  public currentPlayerIndex!: number;
+  private currentWinnerIndex!: number;
+  private playerNames: string[];
+  public boardContent!: number[][];
 
-  public drop(colIx: number) {
-    console.log(`Coin dropped in column ${colIx}`);
-    this.currentPlayerIndex = this.currentPlayerIndex === 1 ? 2 : 1;
+  constructor() {
+    this.playerNames = ['', '1', '2'];
+
+    this.onRestart();
+  }
+
+  onRestart() {
+    this.boardContent = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.currentPlayerIndex = 1;
+    this.currentWinnerIndex = 0;
+  }
+
+  public getPlayerName(col: number, row: number): string {
+    return this.playerNames[this.boardContent[row][col]];
+  }
+
+  public getStyle(col: number, row: number): string {
+    if (this.boardContent[row][col] !== 0) {
+      return `occupied-${this.getPlayerName(col, row)}`; // Like StringFormat in Java
+    }
+
+    return '';
+  }
+
+  public drop(colIx: number): void {
+    if (this.currentWinnerIndex !== 0) {
+      return;
+    }
+
+    for (let i = this.boardContent[colIx].length - 1; i >= 0; i--) {
+      if (this.boardContent[colIx][i] === 0) {
+        this.boardContent[colIx][i] = this.currentPlayerIndex;
+        this.currentPlayerIndex = this.currentPlayerIndex === 1 ? 2 : 1;
+        this.getWinnerIndex();
+        break;
+      }
+    }
+  }
+
+  public get winnerIndex(): number {
+    return this.currentWinnerIndex;
+  }
+
+  public getWinnerName(): string {
+    return this.playerNames[this.currentWinnerIndex];
   }
 
+  /**
+   * get the player (1 or 2) who has won the game
+   */
+  private getWinnerIndex(): void {
```

Comment:
Minor: The method does not return the winner -> `getXXX` is not a good name. Something like `RefreshWinnerIndex` would be better.